__tests__/wallet/sendTransactionWalletService.test.ts

```diff
@@ -779,4 +779,416 @@ describe('prepareTxData', () => {
+          txId: 'wrong-mapping',
+          index: 0,
+          value: 10n,
+          address: 'some-address',
+          tokenId: '02', // Different from output
+          authorities: 0,
+          addressPath: "m/44'/280'/0'/0/1",
+        };
+      }
+      return null;
+    });
+
+    const inputs = [{ txId: 'wrong-mapping', index: 0 }];
+    const outputs = [
+      {
+        type: OutputType.P2PKH,
+        address: 'WP1rVhxzT3YTWg8VbBKkacLqLU2LrouWDx',
+        value: 10n,
+        token: '01',
+      },
+    ];
+
+    sendTransaction = new SendTransactionWalletService(wallet, { inputs, outputs });
+
+    await expect(sendTransaction.prepareTxData()).rejects.toThrow(
+      'Invalid input selection. Input wrong-mapping at index 0 has token 02 that is not on the outputs.'
+    );
+  });
+
+  it('should correctly set fullTxData and _utxosAddressPath properties', async () => {
+    const mockIsValid = jest.spyOn(Address.prototype, 'isValid');
+    mockIsValid.mockReturnValue(true);
+
+    const mockGetType = jest.spyOn(Address.prototype, 'getType');
+    mockGetType.mockReturnValue('p2pkh');
+
+    wallet.getUtxoFromId.mockImplementation(async (txId, index) => {
+      if (txId === 'test-tx' && index === 0) {
+        return {
+          txId: 'test-tx',
+          index: 0,
+          value: 10n,
+          address: 'some-address',
+          tokenId: '01',
+          authorities: 0,
+          addressPath: "m/44'/280'/0'/0/1",
+        };
+      }
+      return null;
+    });
+
+    const inputs = [{ txId: 'test-tx', index: 0 }];
+    const outputs = [
+      {
+        type: OutputType.P2PKH,
+        address: 'WP1rVhxzT3YTWg8VbBKkacLqLU2LrouWDx',
+        value: 10n,
+        token: '01',
+      },
+    ];
+
+    sendTransaction = new SendTransactionWalletService(wallet, { inputs, outputs });
+    const txData = await sendTransaction.prepareTxData();
+
+    // Check that the instance properties are set
+    expect(sendTransaction.fullTxData).toBe(txData);
+    expect(sendTransaction._utxosAddressPath).toEqual(["m/44'/280'/0'/0/1"]);
+  });
+
+  it('should use changeAddress if provided instead of generating new one', async () => {
+    const mockIsValid = jest.spyOn(Address.prototype, 'isValid');
+    mockIsValid.mockReturnValue(true);
+
+    const mockGetType = jest.spyOn(Address.prototype, 'getType');
+    mockGetType.mockReturnValue('p2pkh');
+
+    wallet.getUtxosForAmount.mockResolvedValue({
+      utxos: [
+        {
+          txId: 'change-test-tx',
+          index: 0,
+          value: 15n,
+          tokenId: '01',
+          address: 'some-address',
+          authorities: 0,
+          addressPath: "m/44'/280'/0'/0/1",
+        },
+      ],
+      changeAmount: 5n,
+    });
+
+    const outputs = [
+      {
+        type: OutputType.P2PKH,
+        address: 'WP1rVhxzT3YTWg8VbBKkacLqLU2LrouWDx',
+        value: 10n,
+        token: '01',
+      },
+    ];
+
+    const customChangeAddress = 'WCustomChangeAddressForTesting';
```

Comment:
thought: We should have a validation if this is:
- A valid address
- If it belongs to this wallet

However, this is not implemented on the other facade, so this shouldn't be done here. Let's open an issue to implement this protection.

Reply:
ok